sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateSequencingContainerModelDao.java

```diff
@@ -30,75 +37,102 @@ public HibernateSequencingContainerModelDao() {
-    return results;
+    QueryBuilder<SequencingContainerModel, SequencingContainerModel> builder =
+        new QueryBuilder<>(currentSession(), SequencingContainerModel.class, SequencingContainerModel.class);
+    builder.addPredicate(
+        builder.getCriteriaBuilder().equal(builder.getRoot().get(SequencingContainerModel_.platformType), platform));
+    builder.addPredicate(builder.getCriteriaBuilder().or(
+        builder.getCriteriaBuilder().like(builder.getRoot().get(SequencingContainerModel_.alias), search + '%'),
+        builder.getCriteriaBuilder().equal(builder.getRoot().get(SequencingContainerModel_.identificationBarcode),
+            search)));
+    return builder.getResultList();
   }
 
   @Override
   public SequencingContainerModel getByPlatformAndAlias(PlatformType platform, String alias) throws IOException {
-    return (SequencingContainerModel) currentSession().createCriteria(SequencingContainerModel.class)
-        .add(Restrictions.eq("platformType", platform))
-        .add(Restrictions.eq("alias", alias))
-        .uniqueResult();
+    QueryBuilder<SequencingContainerModel, SequencingContainerModel> builder =
+        new QueryBuilder<>(currentSession(), SequencingContainerModel.class, SequencingContainerModel.class);
+    builder.addPredicate(
+        builder.getCriteriaBuilder().equal(builder.getRoot().get(SequencingContainerModel_.platformType), platform));
+    builder.addPredicate(
+        builder.getCriteriaBuilder().equal(builder.getRoot().get(SequencingContainerModel_.alias), alias));
+    return builder.getSingleResultOrNull();
   }
 
   @Override
-  public SequencingContainerModel getByPlatformAndBarcode(PlatformType platform, String identificationBarcode) throws IOException {
-    return (SequencingContainerModel) currentSession().createCriteria(SequencingContainerModel.class)
-        .add(Restrictions.eq("platformType", platform))
-        .add(Restrictions.eq("identificationBarcode", identificationBarcode))
-        .uniqueResult();
+  public SequencingContainerModel getByPlatformAndBarcode(PlatformType platform, String identificationBarcode)
+      throws IOException {
+    QueryBuilder<SequencingContainerModel, SequencingContainerModel> builder =
+        new QueryBuilder<>(currentSession(), SequencingContainerModel.class, SequencingContainerModel.class);
+    builder.addPredicate(
+        builder.getCriteriaBuilder().equal(builder.getRoot().get(SequencingContainerModel_.platformType), platform));
+    builder.addPredicate(builder.getCriteriaBuilder()
+        .equal(builder.getRoot().get(SequencingContainerModel_.identificationBarcode), identificationBarcode));
+    return builder.getSingleResultOrNull();
   }
 
   @Override
   public long getUsage(SequencingContainerModel model) throws IOException {
-    return getUsageBy(SequencerPartitionContainerImpl.class, "model", model);
+    return getUsageBy(SequencerPartitionContainerImpl.class, SequencerPartitionContainerImpl_.MODEL, model);
```

Comment:
Prefer the `SingularAttribute` version of this and similar methods that you added when possible (`SequencerPartitionContainerImpl_.model` instead of `.MODEL`)